airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -2228,6 +2233,157 @@ public void writeBuilderProject(final ConnectorBuilderProject builderProject) th
+   * @throws ConfigNotFoundException exception if no match on DECLARATIVE_MANIFEST.ACTOR_DEFINITION_ID
+   *         and DECLARATIVE_MANIFEST.VERSION
+   */
+  public DeclarativeManifest getDeclarativeManifestByActorDefinitionIdAndVersion(final UUID actorDefinitionId, final long version)
+      throws IOException, ConfigNotFoundException {
+    final Optional<DeclarativeManifest> declarativeManifest = database
+        .query(ctx -> ctx
+            .select(DECLARATIVE_MANIFEST.asterisk())
+            .from(DECLARATIVE_MANIFEST)
+            .where(DECLARATIVE_MANIFEST.ACTOR_DEFINITION_ID.eq(actorDefinitionId).and(DECLARATIVE_MANIFEST.VERSION.eq(version)))
+            .fetch())
+        .map(DbConverter::buildDeclarativeManifest)
+        .stream().findFirst();
+    return declarativeManifest.orElseThrow(() -> new ConfigNotFoundException(ConfigSchema.DECLARATIVE_MANIFEST,
+        String.format("actorDefinitionId:%s,version:%s", actorDefinitionId, version)));
+  }
+
+  /**
+   * Read currently active declarative manifest by actor definition id by joining with
+   * active_declarative_manifest for the same actor definition id with manifest
+   *
+   * @param actorDefinitionId actor definition id
+   * @throws IOException exception while interacting with db
+   * @throws ConfigNotFoundException exception if no match on DECLARATIVE_MANIFEST.ACTOR_DEFINITION_ID
+   *         that matches the version of an active manifest
+   */
+  public DeclarativeManifest getCurrentlyActiveDeclarativeManifestsByActorDefinitionId(final UUID actorDefinitionId)
+      throws IOException, ConfigNotFoundException {
+    Optional<DeclarativeManifest> declarativeManifest = database
+        .query(ctx -> ctx
+            .select(DECLARATIVE_MANIFEST.asterisk())
+            .from(DECLARATIVE_MANIFEST)
+            .join(ACTIVE_DECLARATIVE_MANIFEST, JoinType.JOIN)
+            .on(DECLARATIVE_MANIFEST.ACTOR_DEFINITION_ID.eq(ACTIVE_DECLARATIVE_MANIFEST.ACTOR_DEFINITION_ID),
+                DECLARATIVE_MANIFEST.VERSION.eq(ACTIVE_DECLARATIVE_MANIFEST.VERSION))
+            .where(ACTIVE_DECLARATIVE_MANIFEST.ACTOR_DEFINITION_ID.eq(actorDefinitionId))
+            .fetch())
+        .map(DbConverter::buildDeclarativeManifest)
+        .stream().findFirst();
+    return declarativeManifest.orElseThrow(() -> new ConfigNotFoundException(ConfigSchema.DECLARATIVE_MANIFEST,
+        String.format("ACTIVE_DECLARATIVE_MANIFEST.actor_definition_id:%s and matching DECLARATIVE_MANIFEST.version", actorDefinitionId)));
+  }
+
+  /**
+   * Read all actor definition ids which have an active declarative manifest pointing to them
+   *
+   * @throws IOException exception while interacting with db
+   */
+  public Stream<UUID> getActorDefinitionIdsWithActiveDeclarativeManifest() throws IOException {
+    return database
+        .query(ctx -> ctx
+            .select(ACTOR_DEFINITION.ID)
+            .from(ACTOR_DEFINITION)
+            .join(ACTIVE_DECLARATIVE_MANIFEST, JoinType.JOIN)
```

Comment:
we discussed to avoid joins between the new tables and the actor definition table - in this case it seems like just checking the actor definition ids from the `ACTIVE_DECLARATIVE_MANIFEST` table is sufficient, so we don't need to do this join.

Reply:
So there are no reason why we might have a ACTIVE_DECLARATIVE_MANIFEST that points to a ACTOR_DEFINITION that has been deleted, right?

Reply:
Actor definitions should never be fully deleted (only soft-deleted via tombstone), so that should be OK